### PR TITLE
feat: add SpriteAnimationData tests

### PR DESCRIPTION
**Pull request by oloshe**
# Description

Add tests for SpriteAnimationData

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

### CODE REVIEW

File: packages/flame/lib/src/sprite_animation.dart [hunk 1]
```diff
@@ -73,6 +73,9 @@ class SpriteAnimationData {
   /// Specifies the range of the sprite grid.
   ///
   /// Make sure your sprites are placed left-to-right and top-to-bottom
+  ///
+  /// [start] start frame index
```
Comment (wolfenrain) [inline packages/flame/lib/src/sprite_animation.dart:77]:
```suggestion
  /// [start] is the start frame index.
```

File: packages/flame/lib/src/sprite_animation.dart [hunk 2]
```diff
@@ -73,6 +73,9 @@ class SpriteAnimationData {
...
+  /// [end] end frame index
```
Comment (wolfenrain) [inline packages/flame/lib/src/sprite_animation.dart:78]:
```suggestion
  /// [end] is the end frame index.
```

**Review by wolfenrain — APPROVED** [review]
LGTM, left a few textual nits

**Comment by oloshe** [comment]
> Awesome that you took the time to add these!

You're welcome. My English maybe not suitable for writing documentation, please point out its blunders if you want.

**Pull request merged**